plenum/common/util.py

```diff
@@ -40,27 +40,32 @@
 Seconds = TypeVar("Seconds", int, float)
 
 
-def randomString(size: int = 20,
-                 chars = string.ascii_letters + string.digits) -> str:
+def randomString(size: int = 20) -> str:
     """
-    Generate a random string of the specified size
+    Generate a random string of the specified size,
+    DONOT use python provided random class its a Pseudo Random Number Generator
+    and not secure enough for our needs
 
     :param size: size of the random string to generate
-    :param chars: the set of characters to use to generate the random string. Uses alphanumerics by default.
     :return: the random string generated
     """
 
-    if not hasattr(chars, "__getitem__"):
-        # choice does not work with non indexed containers
-        chars = list(chars)
+    def randomStr(size):
+        assert (size > 0), "Expected random string size cannot be less than 1"
+        rstr = randombytes(size).hex()
```

Comment:
I agree that it's more costly in case of odd size, but we need it anyway for seeds only for production usage, other usages are test, so as far as production code is concerned, this code is not called that frequently, i don't agree that its complicated, but there is no objective scale to measure "complicated" so we have reached an impasse :)